### PR TITLE
cc12 attempt

### PR DESCRIPTION
**Pull request by nojadzworld**
- [x ] Top-level README “Table of Contents” is updated
 - [x ] README for this challenge is complete
       - [ x] Summary, Description, Approach & Efficiency, Solution
       - [ x] Picture of whiteboard
       - [x] Link to code
 - [ x] Feature tasks for this challenge are completed
 - [ ] Unit tests written and passing
       - [ ] “Happy Path” - Expected outcome
       - [ ] Expected failure
       - [ ] Edge Case (if applicable/obvious)

### CODE REVIEW

File: java/datastructures/lib/src/main/java/codechallenges/linkedlist/AnimalShelter.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  // Dequeue method
+  public Animal dequeue(AnimalType pref) {
+    if (pref == AnimalType.DOG && !dogQueue.isEmpty()) {
+      return dogQueue.dequeue();
+    } else if (pref == AnimalType.CAT && !catQueue.isEmpty()) {
+      return catQueue.dequeue();
+    } else if (pref == null) {
+      // Return the oldest animal if no preference is specified
+      if (!dogQueue.isEmpty() && !catQueue.isEmpty()) {
+        if (dogQueue.peek().getName().compareTo(catQueue.peek().getName()) < 0) {
```
Comment (lucy-gelderloos) [inline java/datastructures/lib/src/main/java/codechallenges/linkedlist/AnimalShelter.java:63]: You're on the right track here, but comparing strings is tricky, and you're going to have to use a lot of workarounds to get the data you want. 

You'll need a way to track the order that each animal was added, independent of any of its other properties. One option would be to add an integer field to your Animal class that tracks the order they were created in. Another would be to redo the AnimalShelter class so it only uses one queue, and find a different way to sort between cats and dogs. I'm sure there are other ways to do it - feel free to experiment!
